analysers/analyser_osmosis_highway_deadend.py

```diff
@@ -194,21 +301,47 @@
   t
 """
 
-sql35 = """
-SELECT
-  DISTINCT ON (oneway.id)
-  oneway.id,
-  oneway.nid,
-  (SELECT ST_AsText(geom) FROM nodes WHERE id = oneway.nid)
+sql42 = """
+CREATE TEMP TABLE results_recursive AS
+SELECT DISTINCT ON (oneway_connection_nodes.wid)
+  oneway_connection_nodes.wid,
+  oneway_connection_nodes.nid
 FROM
-  oneway
+  oneway_connection_nodes
   LEFT JOIN r ON
-    r.nid = oneway.nid
+    r.id = oneway_connection_nodes.nid
 WHERE
-  r.nid IS NULL
+  r.id IS NULL
+ORDER BY
+  oneway_connection_nodes.wid
 """
 
-sql40 = """
+sql43 = """
+SELECT DISTINCT ON(nid)
+  wid,
+  nid,
+  ST_AsText(geom)
+FROM (
+  SELECT
+    wid,
+    nid,
+    geom
+  FROM
+    results_straightforward
+  UNION ALL
+  SELECT
+    wid,
+    nid,
+    (SELECT geom FROM nodes WHERE id = nid) AS geom
```

Comment:
Better use a join.

Reply:
Sure, this was just because it was already like that in the original implementation ( https://github.com/osm-fr/osmose-backend/blob/dev/analysers/analyser_osmosis_highway_deadend.py#L202 ), so I didn't change it, but in terms of execution it's almost equivalent:

`(SELECT geom FROM nodes WHERE id = nid) AS geom` ->
```
['Unique  (cost=10538.70..10553.30 rows=200 width=48) (actual time=0.315..0.362 rows=205 loops=1)']
['  ->  Sort  (cost=10538.70..10546.00 rows=2920 width=48) (actual time=0.314..0.326 rows=205 loops=1)']
['        Sort Key: results_straightforward.nid, results_straightforward.wid']
['        Sort Method: quicksort  Memory: 47kB']
['        ->  Result  (cost=0.00..10370.62 rows=2920 width=48) (actual time=0.042..0.259 rows=205 loops=1)']
['              ->  Append  (cost=0.00..6691.42 rows=2920 width=48) (actual time=0.040..0.137 rows=205 loops=1)']
['                    ->  Seq Scan on results_straightforward  (cost=0.00..20.70 rows=1070 width=48) (actual time=0.039..0.066 rows=202 loops=1)']
['                    ->  Seq Scan on results_recursive  (cost=0.00..6656.12 rows=1850 width=48) (actual time=0.028..0.053 rows=3 loops=1)']
['                          SubPlan 1']
['                            ->  Index Scan using pk_nodes on nodes  (cost=0.56..3.58 rows=1 width=32) (actual time=0.014..0.015 rows=1 loops=3)']
['                                  Index Cond: (id = results_recursive.nid)']
['Planning Time: 0.180 ms']
['Execution Time: 0.442 ms']
```

`JOIN nodes ON nodes.id = results_recursive.nid` ->
```
['Unique  (cost=10534.70..10549.30 rows=200 width=48) (actual time=0.306..0.354 rows=205 loops=1)']
['  ->  Sort  (cost=10534.70..10542.00 rows=2920 width=48) (actual time=0.305..0.318 rows=205 loops=1)']
['        Sort Key: results_straightforward.nid, results_straightforward.wid']
['        Sort Method: quicksort  Memory: 47kB']
['        ->  Result  (cost=0.00..10366.63 rows=2920 width=48) (actual time=0.053..0.246 rows=205 loops=1)']
['              ->  Append  (cost=0.00..6687.43 rows=2920 width=48) (actual time=0.051..0.134 rows=205 loops=1)']
['                    ->  Seq Scan on results_straightforward  (cost=0.00..20.70 rows=1070 width=48) (actual time=0.050..0.078 rows=202 loops=1)']
['                    ->  Nested Loop  (cost=0.56..6633.63 rows=1850 width=48) (actual time=0.019..0.039 rows=3 loops=1)']
['                          ->  Seq Scan on results_recursive  (cost=0.00..28.50 rows=1850 width=16) (actual time=0.002..0.002 rows=3 loops=1)']
['                          ->  Index Scan using pk_nodes on nodes  (cost=0.56..3.57 rows=1 width=40) (actual time=0.010..0.010 rows=1 loops=3)']
['                                Index Cond: (id = results_recursive.nid)']
['Planning Time: 0.206 ms']
['Execution Time: 0.449 ms']
```